src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -361,3 +429,60 @@ def generate_grant_sql_ddl_statements(
         grant_sql_statements.append(grant_sql_statement)
 
     return "\n".join(grant_sql_statements)
+
+
+def generate_new_sql_file_name(
+    deploy_root: Path, generated_root: Path, py_file: Path
+) -> Path:
+    """
+    Generates a SQL filename for the generated root from the python file, and creates its parent directories.
+    """
+    relative_py_file = py_file.relative_to(deploy_root)
+    sql_file = Path(generated_root, relative_py_file.with_suffix(".sql"))
+    if sql_file.exists():
+        cc.warning(
+            f"""\
+            File {sql_file} already exists, will append SQL statements to this file.
+        """
+        )
```

Comment:
Impossible path?

Reply:
Same as one of my comments above. This is technically not possible unless a user themselves creates a `__generated` folder in deploy root, for whatever reason. And then checked in a `py_file_name.sql` in that folder. 
Highly unlikely they do that, but that's why the warning and the appending to write to the file instead of overwriting it entirely.

Reply:
One mismatch currently in this code:
If the main __generated.sql exists, I flag that error and return, not appending.
But if the generated subfiles exists, I append to those. 
We can make this behavior consistent. I dont have a strong opinion either way, so WDYT?

Reply:
See above, I think we stop generation if the user has the own __generated dir (or whatever the config value is set to). We preserve the invariant that we generate to an empty directory. If there's a clash, it means that something's off with symlinks or something like that, I'd probably want an error message to figure out what's going on. It's very much unexpected.